python_modules/dagster-graphql/dagster_graphql/implementation/pipeline_execution_manager.py

```diff
@@ -166,12 +166,16 @@ def __init__(self):
         self._processing_semaphore = gevent.lock.Semaphore(0)
 
         gevent.spawn(self._start_polling)
+        atexit.register(self._cleanup)
 
     def _start_polling(self):
         while True:
             self._poll()
             gevent.sleep(0.1)
 
+    def _cleanup(self):
+        self.join()
```

Comment:
this forces the parent to wait for its children to die before it exits, giving it time to receive the failure events